### PR TITLE
Added skill range and input feedback

### PR DESCRIPTION
**Pull request by saugon**
### What's new
- Added a skill range. The goal is to give feedback to players when they hold down a skill button. (1)
- Refactored the skill input feedback:
  - Improved directional effects and assets. (3)
  - Improved area assets. (6)
  - Added a "self area of effect" when using skills that affects only the player that executes it. (5)
- Each character has its feedback color.
- The feedback color changes to red when dragging over the cancel area. (7)
- Directional skills now can be set as "quick tap". That allows the skills to be executed on tap, towards the player orientation. (4)

### To improve later
- Better directional and area assets.
- Improve 'self area of effect' and how it's affected by the position feedback (the circle below the character).

### References
1. Skill range. Tap feedback:
https://github.com/lambdaclass/curse_of_myrra/assets/6811225/73789e24-e28c-4ec6-894f-a619089d617d

2. Cone feedback.
https://github.com/lambdaclass/curse_of_myrra/assets/6811225/0502c051-f7b0-4cf5-af21-f60133492424

3. Arrow feedback:
https://github.com/lambdaclass/curse_of_myrra/assets/6811225/be6d323b-8905-4616-b01b-574f609c3a43

4. Quick tap (the first skill is 'quick tap'):
https://github.com/lambdaclass/curse_of_myrra/assets/6811225/e0295040-1f79-4519-abff-a6bd6129d6df

5. Self effect:
https://github.com/lambdaclass/curse_of_myrra/assets/6811225/0c2cf472-db0a-4870-8099-f0977756b9d4

6. Area effect:
https://github.com/lambdaclass/curse_of_myrra/assets/6811225/0c63beaa-42ba-426a-81d1-2808df745d73

7. Cancel:
https://github.com/lambdaclass/curse_of_myrra/assets/6811225/163a0b4b-b741-4800-b3ce-fea084355c2b

### CODE REVIEW

**Review by mgrunwaldt — CHANGES_REQUESTED** [review]
Great work! Just a few comments / refactors / typos

File: client/Assets/Scripts/AimDirection.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+        if (skill.GetIndicatorType() == UIIndicatorType.Arrow)
+        {
+            float scaleX = skill.GetArroWidth();
+            float scaleY = skill.GetSkillRadius();
+            arrow.transform.localScale = new Vector3(scaleX, scaleY, 0.05f);
+            arrow.transform.localPosition = new Vector3(0, -scaleY / 2, 0);
+        }
+
+        surface.transform.localScale = new Vector3(viewDistance * 2, viewDistance * 2, 0.05f);
```
[the remark this answers is not in the document]
Comment (BertovDev) [reply client/Assets/Scripts/AimDirection.cs:54]: Also I believe that the 0.05f value is affecting nothing. I changed to a different value and nothing has changed.

File: client/Assets/Scripts/AimDirection.cs [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        characterFeedbackColor = color;
+        initialPosition = transform.localPosition;
+
+        SetColor(color);
+
+        if (skill.GetIndicatorType() == UIIndicatorType.Arrow)
+        {
+            float scaleX = skill.GetArroWidth();
+            float scaleY = skill.GetSkillRadius();
+            arrow.transform.localScale = new Vector3(scaleX, scaleY, 0.05f);
```
[the remark this answers is not in the document]
Comment (BertovDev) [reply client/Assets/Scripts/AimDirection.cs:50]: I believe the `0.05f` it is used to the scaleZ of the image to make it look flat. This value adjust the image to looks good depending of the perspective of the camera. We really can use any value between 0.01 and maybe 0.3

File: client/Assets/Scripts/AimDirection.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            arrow.transform.localPosition = new Vector3(0, -scaleY / 2, 0);
+        }
+
+        surface.transform.localScale = new Vector3(viewDistance * 2, viewDistance * 2, 0.05f);
+        surface.GetComponentInChildren<Renderer>().material.color = new Color32(255, 255, 255, 100);
```
[the remark this answers is not in the document]
Comment (BertovDev) [reply client/Assets/Scripts/AimDirection.cs:55]: We can create an static Enum of colors in the `Utils.cs` to reuse them whenever we want.

File: client/Assets/Scripts/CustomInputManager.cs [hunk 4]
```diff
@@ -124,6 +124,12 @@ protected override void Start()
         UIControlsWrapper.GetComponent<CanvasGroup>().alpha = 0;
     }
 
+    public void Setup()
+    {
+        _player = Utils.GetPlayer(SocketConnectionManager.Instance.playerId);
+        directionIndicator = _player.transform.Find("AimDirection").GetComponent<AimDirection>();
```
Comment (BertovDev) [inline client/Assets/Scripts/CustomInputManager.cs:130]: We can simplify this by doing:
```suggestion
_player.GetComponentInChildren<AimDirection>();
```

**Review by mgrunwaldt — DISMISSED** [review]

**Review by BertovDev — APPROVED** [review]

**Pull request merged**